### DISCUSSION TITLE
linter-js-standard-engine

### DISCUSSION THREAD

**Discussion by ghost** [pulsar-edit/.github#184] [Community Package Help] [answered]
I'd like to register interest in the [`linter-js-standard-engine`](https://github.com/gustavnikolaj/linter-js-standard-engine) package. It's JavaScript linting with project-controlled linter version. The alternative package [`linter-js-standard`](https://github.com/ricardofbarros/linter-js-standard) is available but it bundles a static old version of the linter (3 major versions behind), so many modern features are unrecognized.

It's currently unmentioned in the Admin Actions list.

(So glad someone continued Atom!)

**Comment by confused-Techie** [comment]
I'm glad you have some interest in those packages, but those are just some community packages, so it's always possible to contact the maintainers of that package as ask for these changes or updates. 

I want to mention the only reason a package would appear in the Admin Actions list is if we have to make changes to it. The Admin Actions list serves as a way to be transparent about events like us removing a package or marking it as broken or deprecated. Not all packages appear there, nor to we want them to. Since if a package appears there, that likely means it's been manually removed, or otherwise we had to mark it as broken.

**Comment by ghost** [comment]
I'm just doing what the docs say :-)

>If one of your favorite packages that you had on Atom is missing feel free to contact us or take a look through the Pulsar Package Registry Administrative Actions Log to see if it's been officially removed.

This package is not in the repo, and Admin Actions doesn't say it was removed.

https://web.pulsar-edit.dev/packages/search?q=linter-js-standard-engine

**Reply by confused-Techie** [reply]
Oh I didn't realize it was missing from the registry totally. My bad, I'll look into why it's not appearing, thanks for letting me know!

**Reply by ghost** [reply]
Thanks so much. I wonder if it was unpublished before APM went down. Not sure if that was even possible.

**Reply by confused-Techie** [reply, marked as answer]
@bojavou So I've looked into this one, and in short my assumption is that the author of this package had unpublished (which yes is supported within the Atom Package Registry, as well as Pulsar's) the package prior to the backend being shut down.

I was able to confirm this be checking a few things:
  * The current Pulsar Package Registry, doesn't have an entry for the package at all. Which it would, even if the package was deleted on the Pulsar Registry. (Only a package's name is saved once deleted, to prevent any malicious Supply Chain Attacks.)
  * I then checked the [logs of the original database migration](https://github.com/confused-Techie/atom-package-migrator-to-db-2/blob/main/log.txt) to see if this package had failed during the initial migration process.
  * Finally, I checked the [log of every package that failed the initial health checks](https://github.com/confused-Techie/atom-package-collection/blob/main/out/non_migrated.json) prior to migration.
 
Considering that this package isn't present in any of those locations, that can only mean that it was never a part of the original archive of packages from Atom. 

This means if you'd like to see it available, I'd recommend forking the repository, and publishing the package itself onto the Pulsar Package Registry. Or you can optionally install it directly from GitHub. Either way please feel free to reach out for any assistance!

**Reply by confused-Techie** [reply]
Also a quite side note, I am glad all of this reminded me that the Admin Actions Log doesn't contain the information about any events that took place before the backend was built, like the original database migration, and health checks from above, so I'll make sure to add those in.

**Reply by ghost** [reply]
Thanks for digging into it!

**Reply by asiloisad** [reply]
fyi install package from github directly: `apm install gustavnikolaj/linter-js-standard-engine`

**Reply by confused-Techie** [reply]
@asiloisad Thanks for pointing this out. Or as long as you aren't on Windows, `pulsar -p gustavnikolaj/linter-js-standard-engine` is recommended
